src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_SIZE = 10;
+    private int size = 0;
+    private int index;
+    private Object[] array = new Object[START_ARRAY_SIZE];
+
```

Comment:
create constructor and initialize `values` with an array of objects of DEFAULT_SIZE, cast it to T[]